scipy/fft/_basic.py

```diff
@@ -1,13 +1,21 @@
 from scipy._uarray import generate_multimethod, Dispatchable
 import numpy as np
 
-def _dispatch(func):
-
-    def replacer(args, kwargs, dispatchables):
-        return args, kwargs
+def _x_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer to replace the transform input array (``x``)
+    """
+    if len(args) > 0:
+        return (dispatchables[0],) + args[1:], kwargs
+    kwargs['x'] = dispatchables[0]
```

Comment:
This modifies the original `dict`, which is fine in theory if always done but not ideal.